tests/tests-hz/src/main/java/com/hazelcast/simulator/tests/map/ComplexSqlPredicateTest.java

```diff
@@ -0,0 +1,545 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.simulator.tests.map;
+
+import com.hazelcast.core.IMap;
+import com.hazelcast.nio.serialization.Portable;
+import com.hazelcast.nio.serialization.PortableReader;
+import com.hazelcast.nio.serialization.PortableWriter;
+import com.hazelcast.query.SqlPredicate;
+import com.hazelcast.simulator.hz.HazelcastTest;
+import com.hazelcast.simulator.test.BaseThreadState;
+import com.hazelcast.simulator.test.annotations.Prepare;
+import com.hazelcast.simulator.test.annotations.Setup;
+import com.hazelcast.simulator.test.annotations.Teardown;
+import com.hazelcast.simulator.test.annotations.TimeStep;
+import com.hazelcast.simulator.tests.map.domain.PortableObjectFactory;
+import com.hazelcast.simulator.worker.loadsupport.Streamer;
+import com.hazelcast.simulator.worker.loadsupport.StreamerFactory;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Random;
+
+import static com.hazelcast.simulator.tests.helpers.HazelcastTestUtils.getOperationCountInformation;
+import static com.hazelcast.simulator.utils.GeneratorUtils.generateString;
+import static com.hazelcast.simulator.utils.GeneratorUtils.generateStrings;
+
+public class ComplexSqlPredicateTest extends HazelcastTest {
+
+    // properties
+    public int keyCount = 1000;
+
+    public int idLength = 26;
+    public int universalIdLength = 16;
+    public int panFirst6 = 999999;
+    public int panLast4 = 9999;
+    public int authCode = 999999;
+
+    private IMap<Integer, Transaction> map;
+
+    private String[] generatedIds;
+
+    @Setup
+    public void setup() {
+        this.map = targetInstance.getMap(name);
+        generatedIds = generateStrings(keyCount, idLength);
```

Comment:
This could lead to a huge amount of data being kept in memory; even though it is just readonly.

Reply:
I understand, but I'm thinking whether it is a problem or how it should be solved. I need to store the IDs somewhere to be able to choose from them.

Reply:
It is a problem which should be solved because it can impact the measurements.

You don't need to store the id's, you can generate them on the fly.

Reply:
Please, just ignore my previous comment, I probably suffered from temporal brain damage. I figured it out.